src/uct/ib/ud/base/ud_ep.c

```diff
@@ -1757,14 +1836,36 @@ void uct_ud_ep_disconnect(uct_ep_h tl_ep)
      * to not invoke user callbacks for the endpoint which was destroyed */
     uct_ud_ep_purge(ep, UCS_ERR_CANCELED);
 
+    if (!(ep->flags & UCT_UD_EP_FLAG_CONNECT_TO_EP) &&
+        (ep->flags & UCT_UD_EP_FLAG_TX)) {
+        /* Remove the endpoint from the connection matching context if it
+         * exists there */
+        uct_ud_iface_cep_remove_ep(iface, ep);
+        ep->flags &= ~UCT_UD_EP_FLAG_TX;
+        uct_ud_ep_ctl_op_del(ep, UCT_UD_EP_OP_CREQ);
+
+        if (uct_ud_ep_is_connected(ep)) {
+            uct_ud_ep_ctl_op_add(iface, ep, UCT_UD_EP_OP_FIN);
+        }
+
+        if (ep->flags & UCT_UD_EP_FLAG_RX) {
+            uct_ud_iface_cep_insert_ready_ep(iface, ep);
+            /* Don't destroy ep if it has RX capability */
+            goto out;
```

Comment:
so no need to set UCT_UD_EP_FLAG_DISCONNECTED flag?

Reply:
yes, we don't need to set `UCT_UD_EP_FLAG_DISCONNECTED` flag here